src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/VSSettings.cs

```diff
@@ -30,33 +34,46 @@ private ISettings SolutionSettings
                     ResetSolutionSettingsIfNeeded();
                 }
 
-                return NuGetUIThreadHelper.JoinableTaskFactory.Run(_solutionSettings.Item2.GetValueAsync);
+                return NuGetUIThreadHelper.JoinableTaskFactory.Run(_solutionSettings!.Item2.GetValueAsync);
             }
         }
 
-        private ISolutionManager SolutionManager { get; set; }
+        private ISolutionManager SolutionManager { get; }
+
+        private IMachineWideSettings? MachineWideSettings { get; }
 
-        private IMachineWideSettings MachineWideSettings { get; set; }
+        private IFileWatcherFactory _fileWatcherFactory;
+        private IFileWatcher _userConfigFileWatcher;
+        private IFileWatcher? _solutionConfigFileWatcher;
 
-        public event EventHandler SettingsChanged;
+        public event EventHandler? SettingsChanged;
 
         public VSSettings(ISolutionManager solutionManager)
             : this(solutionManager, machineWideSettings: null)
         {
         }
 
         [ImportingConstructor]
-        public VSSettings(ISolutionManager solutionManager, IMachineWideSettings machineWideSettings)
+        public VSSettings(ISolutionManager solutionManager, IMachineWideSettings? machineWideSettings)
```

Comment:
Why is only IMachineWideSettings being flagged for nullable? 
Looks like the analyzers are wrong here.

Reply:
No, the current code designed like this:

https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/VSSettings.cs#L43-L52

Given there were no tests for this class before this PR, and zero references to the one parameter constructor, I can't figure out why it was designed like this. I can change it.

Reply:
Oh now it makes sense. 

Given that it's not an importing constructor, I'm fairly confident it's only being used in tests so changing it would be cheap/free. Up to you whether you make that change.